File: 백준/Silver/11723. 집합/집합.py
import sys 

M = int(sys.stdin.readline())
S=set()

for _ in range(M):
    temp = sys.stdin.readline().strip().split()
    
    if len(temp)==1:
        op = temp[0]
        if op == "all":    
            S = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20}
        elif op == "empty":
            S = set()    
    else:
        op,num = temp[0],temp[1]
        num = int(num)
        if op == "check":
            if num in S:
                print(1)
            else:
                print(0)
        elif op == "add":
            S.add(num)
        elif op == "remove":
            S.discard(num)
        elif op == "toggle":
            if num in S:
                S.discard(num)
            else:
                S.add(num)

        
        

File: 백준/Silver/11723. 집합/집합.java
import java.io.*; 
import java.util.*; 

public class Main {
	static StringBuilder sb = new StringBuilder(); 

	public static void operation(HashSet<Integer> set,String oper,int target) {
		switch(oper) {
			case "add": 
				set.add(target);
				break; 
			case "check" : 
				if(set.contains(target)) {sb.append(1+"\n");}
				else {sb.append(0+"\n");}
				break; 
			case "remove" : 
				set.remove(target); 
				break; 
			case "toggle" : 
				if(set.contains(target)) {set.remove(target);}
				else {set.add(target); }
				break; 
		
		}
	}
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		HashSet<Integer> set = new HashSet<>(); 
		
		int M = Integer.parseInt(br.readLine()); 
		for(int i=0;i<M;i++) {
			StringTokenizer st = new StringTokenizer(br.readLine()); 
			String oper = st.nextToken(); 
			if(oper.equals("all")){
				for(int j=1;j<21;j++) {
					set.add(j);}
			}
			else if(oper.equals("empty")) {	
				set.clear();
			}
			else {
				int target = Integer.parseInt(st.nextToken()); 
				operation(set,oper,target);
			}
		}
		System.out.println(sb);
		br.close();
	}

}
